tests/framework/filters/AccessRuleTest.php

```diff
@@ -180,10 +180,28 @@ public function testMatchRole($actionid, $allow, $userid, $expected) {
         $this->assertEquals($expected, $rule->allows($action, $user, $request));
     }
 
+    /**
+     * Test that matching role is not possible without User component
+     *
+     * @see https://github.com/yiisoft/yii2/issues/4793
+     */
+    public function testMatchRoleWithoutUser() {
+        $action = $this->mockAction();
+        $request = $this->mockRequest();
+
+        $rule = new AccessRule([
+            'allow' => true,
+            'roles' => ['@'],
+        ]);
+
+        $this->setExpectedException('yii\base\InvalidConfigException');
```

Comment:
Should be replaced with `expectException()`

https://github.com/yiisoft/yii2/blob/master/tests/compatibility.php#L18